duckduckgo/macos-browser
# Phishing Detection decidePolicy handler

Task/Issue URL: https://app.asana.com/0/0/1208206015949665/f
Tech Design URL: https://app.asana.com/0/481882893211075/1207220724600204/f
CC:

**Description**:
Implement URL checking + redirecting to duck://error page for phishing detection.

**Steps to test this PR**:
1. Visit privacy-test-pages.site/security/badware/phishing.html
2. Ensure the browser navigates to a Duck error page, i.e. address bar says: `duck://error?reason=phishing&url=aHR0cDovL3ByaXZhY3ktdGVzdC1wYWdlcy5zaXRlL3NlY3VyaXR5L2JhZHdhcmUvcGhpc2hpbmcuaHRtbA&token=b2G1T-qOqTjZXNBAVYB6qSqOGEdTpXjQpvh4XjLKCDE:1725446933`



**Definition of Done**:


---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
